packages/common/src/utils/charts.ts

```diff
@@ -0,0 +1,64 @@
+import {
+    type CreateSavedChartVersion,
+    type ReplaceCustomFields,
+    type SkippedReplaceCustomFields,
+} from '../types/savedCharts';
+import { getItemId } from './item';
+
+export function maybeReplaceFieldsInChartVersion({
```

Comment:
why is this common ?

Reply:
We need to have the same logic in the explore page/provider. 
When creating/editing a chart, if the explore has a metric that matches a custom metric we replace it in memory.